lib/hyrax/transactions/apply_change_set.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+require 'hyrax/transactions/transaction'
+
+module Hyrax
+  module Transactions
+    ##
+    # Applies and saves a ChangeSet.
+    #
+    # @since 3.0.0
+    #
+    # @example Applying a ChangeSet to a Work
+    #   work = Hyrax::Work.new
+    #   change_set = Hyrax::ChangeSet.for(work)
+    #   change_set.title = ['Comet in Moominland']
+    #
+    #   transaction = Hyrax::Transactions::ApplyChangeSet.new
+    #   persisted   = transaction.call(change_set).value_or(:error)
+    #
+    #   persisted.persisted? # => true
+    #   persisted.title      # => ['Comet in Moominland']
+    #
+    class ApplyChangeSet < Transaction
+      DEFAULT_STEPS = ['change_set.set_modified_date',
+                       'change_set.set_uploaded_date',
```

Comment:
I understand the "change_set.set_modified_date", "change_set.validate", and "change_set.save"; However the "change_set.set_uploaded_date" doesn't quite make sense. Perhaps further reading will help.

Reply:
maybe set `set_uploaded_date` could be renamed? it's actually something like `set_uploaded_date_if_not_present`. the transaction assumes you'll never want to set the uploaded date if it has already been set.

the workflow is:
  - set the modified date to now;
  - set the uploaded date to the modified date, if not already set
  - ...

Reply:
The `set_uploaded_date_if_not_present` sounds more descriptive. I prefer that.

Reply:
renamed!